src/layers/core/scatterplot-layer/scatterplot-layer.js

```diff
@@ -98,10 +110,17 @@ export default class ScatterplotLayer extends Layer {
         })
       ),
       geometry: new Geometry({
-        drawMode: 'TRIANGLE_FAN',
+        drawMode: drawMode || 'TRIANGLE_FAN',
         positions: new Float32Array(positions)
       }),
-      isInstanced: true
+      isInstanced: true,
+      onBeforeRender() {
```

Comment:
In deck.gl v3 it is possible to override Layer.draw(). That enables cleaner handling of gl parameters compared to setting state and using callbacks on the model
 See [ArcLayer](https://github.com/uber/deck.gl/blob/dev/src/layers/core/arc-layer/arc-layer.js#L70).